string action_str
string[] action_names
int32[] action_ids
uint8 selected_action
uint8 selected_step
string[] object_names